Model/Service/Category/CategoryMappingService.php

```diff
@@ -0,0 +1,139 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2020 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Model\Service\Category;
+
+use Exception;
+use Magento\Catalog\Model\Category;
+use Magento\Catalog\Model\ResourceModel\Category\CollectionFactory;
+use Magento\Store\Model\Store;
+use Nosto\Cmp\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Logger\Logger;
+use Nosto\Tagging\Model\Service\Product\Category\DefaultCategoryService as CategoryBuilder;
+
+class CategoryMappingService implements CategoryMappingServiceInterface
+{
+
+    /** @var CollectionFactory */
+    private $collectionFactory;
+
+    /** @var CategoryBuilder */
+    private $categoryBuilder;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var Logger */
+    private $logger;
+
+    /**
+     * @param CollectionFactory $collectionFactory
+     * @param CategoryBuilder $categoryBuilder
+     * @param NostoHelperData $nostoHelperData
+     * @param Logger $logger
+     */
+    public function __construct(
+        CollectionFactory $collectionFactory,
+        CategoryBuilder $categoryBuilder,
+        NostoHelperData $nostoHelperData,
+        Logger $logger
+    ) {
+        $this->collectionFactory = $collectionFactory;
+        $this->categoryBuilder = $categoryBuilder;
+        $this->nostoHelperData = $nostoHelperData;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Store $store
+     * @return string
+     */
+    public function getCategoryMapping(Store $store): string
+    {
+        $array = $this->getMagentoCategories($store);
+        $mapping = json_encode((object)$array, JSON_UNESCAPED_SLASHES);
+        if ($mapping == false) {
+            return '';
```

Comment:
What's up here? If the result is false i.e. malformed - you swallow it and return an empty string?